lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -467,14 +468,25 @@ func getHelmSettings() (*helmCli.EnvSettings, error) {
 
 // checkAgentAlreadyInstalled checks through the Helm if teleport-kube-agent chart was already installed in the EKS cluster.
 func checkAgentAlreadyInstalled(actionConfig *action.Configuration) (bool, error) {
-	listCmd := action.NewList(actionConfig)
-
-	releases, err := listCmd.Run()
+	var releases []*release.Release
+	var err error
+	// We setup a little backoff loop because sometimes access entry auth needs a bit more time to propagate and take
+	// effect, so we could get errors when trying to access cluster right after giving us permissions to do so.
+	for attempt := 1; attempt <= 3; attempt++ {
+		listCmd := action.NewList(actionConfig)
+		releases, err = listCmd.Run()
+		if err != nil {
```

Comment:
Is there a way to tell if the error actually stems from the permission propagation issue and only loop in this case?

Reply:
I saw at least two different type of errors that can occur and I'm not sure if some other can occur as well, so I think it's better to retry just on any error.